src/main/java/hudson/plugins/cigame/UserScorePropertyDescriptor.java

```diff
@@ -44,10 +49,16 @@ private UserScoreProperty newInstanceIfJSONIsNull(StaplerRequest req) throws For
 
     @Override
     public UserScoreProperty newInstance(StaplerRequest req, JSONObject formData) throws hudson.model.Descriptor.FormException {
+
         if (formData == null) {
             return newInstanceIfJSONIsNull(req);
         }
         if (formData.has("score")) { //$NON-NLS-1$
+            if (getCurrentUserScore() != getRequestScore(formData.get("score").toString())) {
+                if (!Hudson.getInstance().getACL().hasPermission(Hudson.ADMINISTER)) {
+                    throw new hudson.model.Descriptor.FormException(Messages.UserScore_Cheating_Message(), "score");
+                }
+            }
```

Comment:
couldn't you extract lines 57-61 and 40-44 into a method? This is an exact code block copy.